packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2945,7 +2952,8 @@ export class ContainerRuntime
 		if (locationInBatch.batchStart) {
 			const firstMessage = messagesWithMetadata[0]?.message;
 			assert(firstMessage !== undefined, 0xa31 /* Batch must have at least one message */);
-			this.scheduleManager.batchBegin(firstMessage);
+			this.deltaScheduler.batchBegin(firstMessage);
```

Comment:
Did you consider having `deltaScheduler` listen to `batchBegin`, rather than CR having to know to call it here?  Only hesitation I have with switching to event-based is that it has a side effect (pausing the queue to breath - its whole point of existence). But the pro of that switch is this code is more concise.

Reply:
Goog suggestion. I had not thought of that.

Reply:
Changed do this on event listeners similar to `InboundBatchAggregator`.